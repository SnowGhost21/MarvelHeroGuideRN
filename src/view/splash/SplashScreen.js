import React, { Component } from 'react';
import { StyleSheet, Image, SafeAreaView } from 'react-native';
import { NavigationActions, StackActions } from 'react-navigation'


export default class SplashScreen extends Component {

    static navigationOptions = ({ navigation }) => {
        return {
            header: () => null
        }
    }

    componentDidMount() {
        const { navigation } = this.props;
        setTimeout(() => {
            navigation.dispatch(StackActions.reset({
                index: 0,
                actions: [
                    NavigationActions.navigate({
                        routeName: 'Home'
                    })
                ]
            }));
        }, 3000);
    }

    render() {
        return (
            <SafeAreaView style={styles.container}>
                <Image source={require('../../../imgs/ic_logo.png')} style={styles.logo} />
            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'red',
    },

    logo: {
        width: 200,
        height: 200,
    },
})