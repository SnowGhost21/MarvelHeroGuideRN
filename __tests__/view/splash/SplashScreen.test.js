import 'react-native';
import React from 'react';
import SplashScreen from '../../../src/view/splash/SplashScreen';
import renderer from 'react-test-renderer';

jest.setTimeout(15000);

it('It should render component correctly', () => {
    const tree = renderer.create(<SplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('It should render image logo correctly', async (done) => {
    const Image = require('Image');
    Image.getSize('../../../imgs/ic_logo.png', (width, height) => {
        const tree = renderer.create(<Image style={{ height, width }} />).toJSON();
        expect(tree).toMatchSnapshot();
        done();
    });
});