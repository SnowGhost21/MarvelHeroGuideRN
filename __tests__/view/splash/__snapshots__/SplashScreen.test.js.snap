// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render component correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../imgs/ic_logo.png",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</RCTSafeAreaView>
`;

exports[`It should render image logo correctly 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../imgs/ic_logo.png",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</RCTSafeAreaView>
`;
